### PR TITLE
Support to run karmadactl init within a pod

### LINKED ISSUE #3335: Failed runing kubectl-karmada within a pod
Met below error when run kubectl-karmada init in a pod. The intention is in [kurator](https://github.com/kurator-dev/kurator) we want to install karmada control plane in a pod. 
```
error: Missing or incomplete configuration info.  Please point to an existing, complete config file:
  1. Via the command-line flag --kubeconfig
  2. Via the KUBECONFIG environment variable
  3. In your home directory as ~/.kube/config

```

I guess it is because of lacking incluster client initiation support.

### PR DESCRIPTION
**Pull request by lonelyCZ**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Support to run karmadactl init within a pod by InClusterConfig.

**Which issue(s) this PR fixes**:
Fixes #3335 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
karmadactl: support to run karmadactl init within a pod
```

### CODE REVIEW

**Comment by lonelyCZ** [comment]
Hi @hzxuzhonghu , Could you please help test it in your env?

File: pkg/karmadactl/util/apiclient/apiclient.go [hunk 1]
```diff
@@ -43,6 +44,13 @@ func RestConfig(context, kubeconfigPath string) (*rest.Config, error) {
 		kubeconfigPath = env.GetString("KUBECONFIG", defaultKubeConfig)
 	}
 	if !Exists(kubeconfigPath) {
+		klog.Warning("Neither --kubeconfig nor KUBECONFIG was specified and '~/.kube/config' doesn't exist.  Using the inClusterConfig.  This might not work.")
+		// if the kubeconfig path is not valid, try to load InClusterConfig
+		inClusterRestConfig, err := rest.InClusterConfig()
```
Comment (hzxuzhonghu) [inline pkg/karmadactl/util/apiclient/apiclient.go:49]: great, will test it

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

It can connect to kube-apiserver successfully, we just need to grant some permission to the service account.

**Comment by lonelyCZ** [comment]
> It can connect to kube-apiserver successfully, we just need to grant some permission to the service account.

Great, I will update the klog message to match our scenario.

**Comment by lonelyCZ** [comment]
If there isn't kubeconfig, the message is that

```
[root@master67 karmada]# _output/bin/linux/amd64/karmadactl init
W0328 17:43:07.952514    1571 apiclient.go:47] Kubeconfig '/root/.kube/config' doesn't exist. Using the inClusterConfig. This might not work.
W0328 17:43:07.953343    1571 apiclient.go:53] error creating inClusterConfig: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined
error: Missing or incomplete configuration info.  Please point to an existing, complete config file:
  1. Via the command-line flag --kubeconfig
  2. Via the KUBECONFIG environment variable
  3. In your home directory as ~/.kube/config
```
What do you think?

/cc @RainbowMango

File: pkg/karmadactl/util/apiclient/apiclient.go [hunk 2]
```diff
@@ -43,6 +44,13 @@ func RestConfig(context, kubeconfigPath string) (*rest.Config, error) {
 		kubeconfigPath = env.GetString("KUBECONFIG", defaultKubeConfig)
 	}
 	if !Exists(kubeconfigPath) {
+		klog.Warningf("Kubeconfig '%s' doesn't exist. Using the inClusterConfig. This might not work.", kubeconfigPath)
```
[the remark this answers is not in the document]
Comment (RainbowMango) [reply pkg/karmadactl/util/apiclient/apiclient.go:47]: How about this?
```golang
	if !Exists(kubeconfigPath) {
		// Given no kubeconfig is provided, give it a try to load the config by
		// in-cluster mode if the client running inside a pod running on kubernetes.
		host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
		if len(host) > 0 || len(port) > 0 { // in-cluster mode
			inClusterRestConfig, err := rest.InClusterConfig()
			if err != nil {
				return nil, fmt.Errorf("failed to load rest config by in-cluster mode: %v", err)
			}
			return inClusterRestConfig, nil
		}

		return nil, ErrEmptyConfig
	}
```

If no kubeconfig is provided and it works inside a Pod, we assume that it is intended to be running inside, so we just return the in-cluster error.

[the remark this answers is not in the document]
Comment (lonelyCZ) [reply pkg/karmadactl/util/apiclient/apiclient.go:47]: I think it is great.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**